content/blog/2019/02/2019-02-21-credentials-masking.adoc

```diff
@@ -0,0 +1,88 @@
+---
+layout: post
+title: Limitations of Credentials Masking
+tags:
+- security
+author: daniel-beck
+---
+
+In the Jenkins project, we ask that people link:/security/#reporting-vulnerabilities[report security issues to our private issue tracker].
+This allows us to review issues and prepare fixes in private, often resulting in better, safer security fixes.
+
+As a side effect of that, we also learn about common misconceptions and usability problems related to security in Jenkins.
+This post is intended to address one of those:
+The goal and limitations of credentials masking.
+
+== The Problem
+
+One very common example of that is the role of credentials masking in Jenkins, typically involving a pipeline snippet that looks like this:
+
+[pipeline]
+----
+// Scripted //
+withCredentials([usernamePassword(credentialsId: 'topSecretCredentials', passwordVariable: 'PWD', usernameVariable: 'USR')])
+  sh './deploy.sh' // requires PWD and USR to be set
+}
+// Declarative //
+----
+
```

Comment:
Is there a reason for no Declarative code?

Reply:
Too lazy 😞  Feel free to propose the declarative equivalent.

Reply:
That explains it ;-)  I'll deal with it when I steal the material for the courseware.  This is a terrific piece -- thanks for writing it up!